_posts/pt_BR/2016-01-01-angularjs-digest-vs-apply.md

```diff
@@ -0,0 +1,36 @@
+---
+layout: post
+
+title: AngularJs - `$digest` vs `$apply`
+tip-number: 01
+tip-username: loverajoel 
+tip-username-profile: https://github.com/loverajoel
+tip-tldr: Módulos JavaScript e ferramentas de build estão ficando cada vez mais numerosas e complicadas, mas e as receitas de bolo nos novos frameworks?
+
+categories:
+    - pt_BR
+---
+
+Uma das características mais apreciadas do AngularJs é o two-way data binding. Para fazer este trabalho o AngularJs avalia as mudanças entre o model e a view através de ciclos ( $digest ). Você precisa entender esse conceito para entender como o framework funciona debaixo do capô.
+
+Angular avalia cada observador quando um evento é disparado. Esse é o ciclo conhecido como `$digest`. Algumas vezes você precisa forçar a execução de um novo ciclo manualmente e deve escolher a opção correta pois esta fase é uma das que mais influencia em termos de performance.
+
+### `$apply`
+Este método do core permite você iniciar o ciclo digest explicitamente. Isso significa que todos os observadores são verificados; toda a aplicação inicia o `$digest loop`. Internamente, depois de executar um parâmetro de função opcional, ele chama `$rootScope.$digest();`.
```

Comment:
I have never heard a programmer talking "observadores" in pt_BR, usually ppl use "watchers". Would "watchers" be better in this context, or maybe something like "watchers (observadores)"?